pex/tools/commands/venv.py

```diff
@@ -206,7 +208,11 @@ def run(self, pex):
                     "installed:\n{}".format(e)
                 )
         if self.options.compile:
-            pex.interpreter.execute(["-m", "compileall", venv_dir])
+            try:
+                pex.interpreter.execute(["-m", "compileall", venv_dir])
+            except Executor.NonZeroExit as non_zero_exit:
+                pex_warnings.warn(non_zero_exit.stderr)
```

Comment:
Have you tested this includes the relevant compile failure information? It would be great if you could add an integration test that uses your case. You could trim your case down to just the problematic [aenum](https://pypi.org/project/aenum/3.1.11/) case.

Integration tests live here: https://github.com/pantsbuild/pex/tree/main/tests/integration
You could use `pex/tests/integration/test_issue_2001.py` for the new test file. Really the test just needs to confirm a venv can be created and check the compilation error shows up in stderr if PEX_VERBOSE=1 (which turns on warnings). Here is a similar enough IT: https://github.com/pantsbuild/pex/blob/31020410809e94fd2a8565294d26b5346164800e/tests/integration/test_issue_1995.py

The key bits are:
+ tmpdir: for making up paths for the test.
+ run_pex_command(...).assert_success(): For running the pex CLI and making sure it works
+ make_env: for creating an env dict that includes the current env + your additions.

Reply:
Thank you John for all the instructions. Yeah I was thinking of adding test but didn't have very good understanding of this repo. I will try to add it soon.